### ISSUE TITLE
Update error: os error 32

### ISSUE THREAD

**Issue by carlocardella** [Microsoft/vscode#55885]
VSCode Insiders has been unable to properly auto-update for the last few releases (daily updates), it started roughly a couple of weeks ago on all my 3 machines. 
I ready through #47841 and its references but I don't think that applies. 

vscode-inno-updater.log is attached. 
[vscode-inno-updater.log](https://github.com/Microsoft/vscode/files/2263114/vscode-inno-updater.log)

I've been able to pinpoint the error to this: 

```
---------------------------
Error
---------------------------
C:\Users\carloc\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\node_modules.asar.unpacked\vscode-ripgrep\bin\rg.exe

An error occurred while trying to replace the existing file:
DeleteFile failed; code 32.
The process cannot access the file because it is being used by another process.

Click Retry to try again, Ignore to skip this file (not recommended), or Abort to cancel installation.
---------------------------
Abort   Retry   Ignore
---------------------------
```

From Process Explorer I see the file is locked by "System" and I cannot close/release the handle. 

I can update VSCode if I kill OneDrive which implies it is somehow locking that file:
![image](https://user-images.githubusercontent.com/5784415/43726595-6c8ebd7e-9954-11e8-9a5e-81b5913bc2da.png)

During my troubleshooting I also found that the auto-update process just fails (it does not offer any option to ignore the error) and leaves VSCode completely unusable. If I then run the full installer I get again an error because rg.exe is locked but at least gives me the option to ignore the error and continue. That does not seem to have averse effects on VSCode, the new build works fine after this type of update, it's just very annoying to go through these steps every day.

Is there anything else I can do? 

_System Info (from my desktop, I can repro on two laptops as well)_
VS Code version: Code - Insiders 1.26.0-insider (c2ddece6536330f4b4ceef33de5eea6e406fb30a, 2018-08-06T10:34:04.876Z)
OS version: Windows_NT x64 10.0.17134

|Item|Value|
|---|---|
|CPUs|Intel(R) Xeon(R) CPU           W3670  @ 3.20GHz (6 x 3200)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: unavailable_software<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|24.00GB (12.04GB free)|
|Process Argv|C:\Users\carloc\AppData\Local\Programs\Microsoft VS Code Insiders\Code - Insiders.exe|
|Screen Reader|no|
|VM|0%|

**Comment by ramya-rao-a** [comment]
cc @roblourens for ripgrep

**Comment by joaomoreno** [comment]
Very interesting that you were able to locate OneDrive as the culprit. Are you sure it is correlated? How did you get to that idea?

**Comment by carlocardella** [comment]
I don't have real proof unfortunately (no trace, or dump I could capture) but since it's been happening for almost two weeks now I could experiment a bit: I noticed a recent change (maybe unrelated, I'm not sure) in the OneDrive notification when a file is being downloaded on-demand:

![r downloading from onedrive](https://user-images.githubusercontent.com/5784415/43783049-9c5045b6-9a15-11e8-8570-49384a8f8371.png)

"r" is downloading was new to me, I'm not sure if it is correlated but since I was experimenting I tried to close OneDrive (both Personal and Business) and VSCode was able to update. I've been doing that for the past few days and the behavior seems fairly reliable.

Interestingly enough though, yesterday and today I could no longer repro the error on one of the two laptops even without closing OneDrive, so something somewhere must have changed but I can't figure out what...

**Comment by carlocardella** [comment]
Mystery solved I think, with tomorrow's insiders update I'll know more. 

I had added a folder sync'ed through OneDrive to my vscode workspace. I'll add that the folder is also a junction, even though I have not tested if that makes any difference:

![image](https://user-images.githubusercontent.com/5784415/43789304-75a179bc-9a24-11e8-8809-7c31c20f1dd7.png)

Today I realized than loading a new file from that "Work" folder caused OneDrive (with the "r is downloading" message I mentioned before) to pop-up, so I removed the OneDrive folder from the workspace, intalled an older vscode version and let it auto-update and this time everything went smooth... 
Important to note: in one of my tabs I still have open a file under the OneDrive folder but that does not seem to have any impact, only OneDrive added as a Workspace folder seems to cause problems. 

![image](https://user-images.githubusercontent.com/5784415/43789503-f950ec20-9a24-11e8-9f9a-8d12c5334700.png)

By the way, this reproes both with Stable and Insider.
